train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -9,6 +9,20 @@
 	private int speedLimit = 0;
 	private Thread thread;
 
+	public TrainControllerImpl() {
+		thread = new Thread() {
+			public void run() {
+				thread.run();
+				try {
+					followSpeed();
+					thread.sleep(1500);
+				} catch(InterruptedException ex) {
+					ex.printStackTrace()	
```

Comment:
Exceptions should be logged, but the team aggreed that for now it is a good solution.